vignettes/sits.Rmd

```diff
@@ -349,6 +349,110 @@ point_cf.tb <- sits_ndvi_arima_filter(point.tb, apply_whit = FALSE)
+
+$$
+D_{j}=\sum_{i=1}^{n}{\sqrt{(x(t)_i{-}w_{ji})^{2}}}.
+$$
+
+The neuron that contains the shortest distance is declared the winner neuron, defined here as ${b}$, also know as Best Matching Unit (BMU):
+
+
+$$
+ d_{b}= min \left\{D_1,\ldots, D_J \right\}.
+$$
+
+The BMU and its neighborhood must be updated. The weights are adjusted to increase the similarity with input vector.
+The equation for updating the weight vector is given by:
+
+$$
+ w_{j}(t{+}1)=  w_{j}(t){+}\alpha(t)* h_{b,j}(t) [x(t)_i{-}w_{j}(t)],
+$$
+where $\alpha(t)$ is the learning rate, and it must be set as $0<\alpha(t)<1$ and $h_{b,j}(t)$ is a neighbourhood function.
+
+In sits, the package kohonen (referencia) is used as basis to run the SOM, after the training step, to separate the samples in groups is necessary to label the neurons. A cluster can be one neuron or a set of neurons that belongs to same category. In this step, a neuron is named using majority vote technique through of original label given from sample data set. As a result of this, a sample that was associated to a neuron receives the same label of it. In some cases, no samples is associated to a neuron, then, it receives the label "Noclass".
+
+```{r}
+data("samples_mt_9classes", package = "sits")
+
+```
+```{r}
+data.tb <- samples_mt_9classes
+```
+
+
+```{r}
+# clustering time series using SOM
+kohonen_cluster <-
+    sits::sits_kohonen(
+        data.tb,
+        bands = c("evi", "ndvi"),
+        grid_xdim = 25,
+        grid_ydim = 25,
+        rlen = 100,
+        distance = "euclidean",
+        neighbourhood.fct = "gaussian",
+        mode = "online"
+
+    )
+# return a tibble with all samples and which cluster it belongs
+samples_cluster <- kohonen_cluster$info_samples
+samples_cluster [1:10,]
+```
+
+To verify the quality of clusters generated by SOM, the confusion matrix can be accessed. From the confusion matrix percentage of mixture within a cluster is shown. 
+
+```{r}
+
+# Calculate the percentage of confusion within a cluster 
+confusion_by_cluster <- sits_evaluate_cluster(kohonen_cluster$info_samples)
+confusion_by_cluster$mixture_cluster[1:5,]
+
+```
+
+```{r}
+# confusion matrix
+confusion_matrix <- confusion_by_cluster$confusion_matrix
+
+sits_plot_cluster_info(confusion_by_cluster, "Confusion by Cluster")
+```
+
+
+Besides that, to evaluate the samples with greater reliability, the process of clustering and labelling can be executed T times. 
```

Comment:
several times instead of T times..
Remove "besides that"
Possibly remove the first sentence.